File: backend/BookstoreAPI/Models/CuotaProveedor.cs
namespace BookstoreAPI.Models
{
    public class CuotaProveedor
    {
        public int Id { get; set; }
        public int ComprobanteProveedor_Id { get; set; }
        public int NumeroCuota { get; set; }
        public DateTime FechaVencimiento { get; set; }
        public decimal Importe { get; set; }
        public decimal ImportePagado { get; set; }
        public string Estado { get; set; } = "PEN";
    }
}

File: frontend/bookstore-app/src/types/cuotaProveedor.ts
export interface PagoCuotaProveedorDto {
  id: number;
  nroReferencia: string;
  fecha: string;
  importe: number;
}

export interface CuotaProveedorListado {
  id: number;
  comprobanteProveedorId: number;
  nroComprobante: string;
  tipoComprobante: string;
  fechaEmision: string;
  proveedorId: number;
  proveedorNombre: string;
  numeroCuota: number;
  fechaVencimiento: string;
  importe: number;
  importePagado: number;
  estado: string;
  pagos: PagoCuotaProveedorDto[];
}

export interface CuotaProveedorFiltros {
  proveedorId?: number;
  fechaDesde?: string;
  fechaHasta?: string;
  estado?: string;
}

export interface CreatePagoCuotaProveedorDto {
  nroReferencia: string;
  fecha: string;
  importe: number;
}
